pkg/cmd/pulumi/auth/login.go

```diff
@@ -205,19 +213,39 @@ func NewLoginCmd() *cobra.Command {
 	return cmd
 }
 
-func validateCloudBackendType(typ string) error {
+func validateCloudBackendType(typ string) (bool, error) {
 	kind := strings.SplitN(typ, ":", 2)[0]
 	supportedKinds := []string{"azblob", "gs", "s3", "file", "https", "http"}
 	for _, supportedKind := range supportedKinds {
 		if kind == supportedKind {
-			return nil
+			if strings.Contains(kind, "http") {
+				return true, nil
+			}
+			return true, nil
 		}
 	}
-	return fmt.Errorf("unknown backend cloudUrl format '%s' (supported Url formats are: "+
+	return false, fmt.Errorf("unknown backend cloudUrl format '%s' (supported Url formats are: "+
 		"azblob://, gs://, s3://, file://, https:// and http://)",
 		kind)
 }
 
+func replaceDomain(input string) string {
+	re := regexp.MustCompile(`(https?://)[^.]+(\.(pulumi\.)?[^.]+\.[^.]+)`)
+
+	// Replace the subdomain with "api"
+	return re.ReplaceAllString(input, "${1}api${2}")
+}
+
+func checkHttpCloudBackenUrl(url string) (string, error) {
+	if strings.HasSuffix(strings.TrimSuffix(url, "/"), "pulumi.com") {
+		return "https://api.pulumi.com", nil
+	}
+	if !strings.Contains(url, "//api.") {
+		return "", fmt.Errorf("did you mean %s ?", replaceDomain(url))
```

Comment:
```suggestion
		return "", fmt.Errorf("did you mean %s?", replaceDomain(url))
```

There should be no space before the `?`.

Reply:
Hi @tgummerer , I removed returning error as it is not valid anymore.
Thanks!